src/FaultCheck.cpp

```diff
@@ -0,0 +1,79 @@
+#include "FaultCheck.hpp"
```

Comment:
SensorReading Class

type sensor_reading
-> value
-> fault status

sfr
sensor_reading gyro_x = 0;

IMUMonitor

..averaging code..

gyro_x.set(averaged value)



MissionManager

if(gyro_x.value() > 4 && gyro_x.is_valid()){}


FaultMonitor

..if out of bounds logic..
gyro_x.set_invalid()